### ISSUE TITLE
test fails on OpenBSD

### ISSUE THREAD

**Issue by bket** [restic/restic#1307]
Regression test fails on OpenBSD with the error below. This error is caused by a comparison in restic/cmd/restic/integration_helpers_unix_test.go:

```
        if !sameModTime(e.fi, other.fi) {
                fmt.Fprintf(os.Stderr, "%v: ModTime does not match (%v != %v)\n", e.path, e.fi.ModTime(), other.fi.ModTime())
                return false
        }
```

It seems that e.fi.ModTime() uses nanosecond accuracy while other.fi.ModTime() uses microsecond accuracy. Comparing the two will result in false, which is negated thus the if-statement will return false.

Is it by design that restic's make test uses two different time resolutions? Is this also the case for restic itself?

```
===>  Regression tests for restic-0.7.3
cd /home/code/ports/pobj/restic-0.7.3/go/src/github.com/restic/restic && /usr/bin/env -i PATH="/home/code/ports/pobj/restic-0.7.3/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/X11R6/bin" TMPDIR="/home/code/ports/pobj/restic-0.7.3/build-amd64" GOPATH="/home/code/ports/pobj/restic-0.7.3/go:/usr/local/go-pkg" PATH="/home/code/ports/pobj/restic-0.7.3/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/X11R6/bin" go run build.go --test
ok  	github.com/restic/restic/internal/archiver	101.494s
ok  	github.com/restic/restic/internal/backend	1.417s
ok  	github.com/restic/restic/internal/backend/azure	0.009s
ok  	github.com/restic/restic/internal/backend/b2	0.009s
ok  	github.com/restic/restic/internal/backend/gs	0.009s
ok  	github.com/restic/restic/internal/backend/local	2.181s
ok  	github.com/restic/restic/internal/backend/location	0.038s
ok  	github.com/restic/restic/internal/backend/mem	1.359s
ok  	github.com/restic/restic/internal/backend/rest	0.009s
ok  	github.com/restic/restic/internal/backend/s3	0.054s
ok  	github.com/restic/restic/internal/backend/sftp	9.276s
ok  	github.com/restic/restic/internal/backend/swift	0.009s
ok  	github.com/restic/restic/internal/backend/test	0.803s
ok  	github.com/restic/restic/internal/checker	6.167s
ok  	github.com/restic/restic/internal/crypto	0.868s
?   	github.com/restic/restic/internal/debug	[no test files]
?   	github.com/restic/restic/internal/errors	[no test files]
ok  	github.com/restic/restic/internal/filter	0.580s
ok  	github.com/restic/restic/internal/fs	0.008s
ok  	github.com/restic/restic/internal/hashing	0.218s
ok  	github.com/restic/restic/internal/index	45.972s
?   	github.com/restic/restic/internal/list	[no test files]
?   	github.com/restic/restic/internal/migrations	[no test files]
?   	github.com/restic/restic/internal/mock	[no test files]
ok  	github.com/restic/restic/internal/options	0.004s
ok  	github.com/restic/restic/internal/pack	0.015s
ok  	github.com/restic/restic/internal/pipe	0.030s
ok  	github.com/restic/restic/internal/repository	50.409s
ok  	github.com/restic/restic/internal/restic	7.016s
?   	github.com/restic/restic/internal/test	[no test files]
ok  	github.com/restic/restic/internal/walk	1.390s
ok  	github.com/restic/restic/internal/worker	0.006s
could not read signature from exclusion tagfile "/home/code/ports/pobj/restic-0.7.3/build-amd64/restic-test-637321099/CACHEDIR.TAG": unexpected EOF
error for testdata/0/0/9/37: Lstat: lstat /home/code/ports/pobj/restic-0.7.3/build-amd64/restic-test-484473378/testdata/0/0/9/37: no such file or directory
error walking dir testdata/0/0/9/37: Lstat: lstat /home/code/ports/pobj/restic-0.7.3/build-amd64/restic-test-527802911/testdata/0/0/9/37: no such file or directory
error walking dir 0/9: Open: open /home/code/ports/pobj/restic-0.7.3/build-amd64/restic-test-118653147/testdata/0/0/9: no such file or directory
 ID          User        Host        Created
----------------------------------------------------------------------
*979ed800    root        gateway.lan  2017-10-01 07:52:54
foo: ModTime does not match (2017-10-01 07:52:55.944998902 +0200 CEST != 2017-10-01 07:52:55.944999 +0200 CEST)
%foo
foo/bar: ModTime does not match (2017-10-01 07:52:56.455001295 +0200 CEST != 2017-10-01 07:52:56.455002 +0200 CEST)
%foo/bar
foo/bar/testfile0: ModTime does not match (2017-10-01 07:52:56.024999151 +0200 CEST != 2017-10-01 07:52:56.025 +0200 CEST)
%foo/bar/testfile0
foo/bar/testfile1: ModTime does not match (2017-10-01 07:52:56.085003878 +0200 CEST != 2017-10-01 07:52:56.085004 +0200 CEST)
%foo/bar/testfile1
foo/bar/testfile2: ModTime does not match (2017-10-01 07:52:56.094998706 +0200 CEST != 2017-10-01 07:52:56.094999 +0200 CEST)
%foo/bar/testfile2
foo/bar/testfile3: ModTime does not match (2017-10-01 07:52:56.165002242 +0200 CEST != 2017-10-01 07:52:56.165003 +0200 CEST)
%foo/bar/testfile3
foo/bar/testfile4: ModTime does not match (2017-10-01 07:52:56.254998437 +0200 CEST != 2017-10-01 07:52:56.254999 +0200 CEST)
%foo/bar/testfile4
foo/bar/testfile5: ModTime does not match (2017-10-01 07:52:56.324999598 +0200 CEST != 2017-10-01 07:52:56.325 +0200 CEST)
%foo/bar/testfile5
foo/bar/testfile6: ModTime does not match (2017-10-01 07:52:56.364998396 +0200 CEST != 2017-10-01 07:52:56.364999 +0200 CEST)
%foo/bar/testfile6
foo/bar/testfile7: ModTime does not match (2017-10-01 07:52:56.37500363 +0200 CEST != 2017-10-01 07:52:56.375004 +0200 CEST)
%foo/bar/testfile7
foo/bar/testfile8: ModTime does not match (2017-10-01 07:52:56.455001295 +0200 CEST != 2017-10-01 07:52:56.455002 +0200 CEST)
%foo/bar/testfile8
foo/bar/testfile9: ModTime does not match (2017-10-01 07:52:56.524998336 +0200 CEST != 2017-10-01 07:52:56.524999 +0200 CEST)
%foo/bar/testfile9
[31mintegration_test.go:905: directories are not equal[39m

--- FAIL: TestRestore (6.35s)
	testing.go:30: using low-security KDF parameters for test
	testing.go:30: using low-security KDF parameters for test
	lock.go:77: setting lock timeout to 0s
	integration_test.go:51: repository initialized at /home/code/ports/pobj/restic-0.7.3/build-amd64/restic-test-301604076/repo
	integration_test.go:55: backing up [/home/code/ports/pobj/restic-0.7.3/build-amd64/restic-test-301604076/testdata]
index d5697fe0 has old format
index d5697fe0 has old format
FAIL
FAIL	github.com/restic/restic/cmd/restic	73.522s
running tests failed: exit status 1
exit status 1
*** Error 1 in . (Makefile:26 'do-test')
```

**Comment by fd0** [comment]
Is it possible that this is a duplicate of #1087?

**Comment by wojas** [comment]
This was resolved in Go with the following issues and commits:

- https://github.com/golang/go/issues/16480 https://github.com/golang/go/commit/67c360594f1cdfa8e663bf5ff51724d3fc31abb9 (utimensat syscall on BSD)
- https://github.com/golang/go/issues/21437 https://github.com/golang/go/commit/0b06929b2deeedc42706598a7ffc95fcbae91ace (syscall param fix)

As suggested in #1087, we should probably add a Go version check to the workaround once a new version of Go has been released that incorporates these fixes.

It looks like this affects all BSDs, not just Darwin and OpenBSD.

**Comment by fd0** [comment]
I think this is fixed by now, at least in Go 1.10. I'm closing this issue for now, please feel free to add further comments. Thanks!
